packages/next/build/webpack/loaders/next-serverless-loader.ts

```diff
@@ -128,6 +128,10 @@ const nextServerlessLoader: loader.Loader = function() {
           .replace(/\\.json$/, '')
       }
       const parsedUrl = parse(req.url, true)
+
+      const { _nextRewrotePathname } = parsedUrl.query
+      delete parsedUrl.query._nextRewrotePathname
```

Comment:
Looks like this won't work for serverless handling since the passed `req.url` is the original so we need to derive the URL by running over the rewrites ourselves here instead

Reply:
Updated to run over the rewrites in the serverless function itself and stopping when we match the page or the dynamic route